### DISCUSSION TITLE
pwnagotchi bluetooth tether GPS termux-location

### DISCUSSION THREAD

**Discussion by krishenriksen** [jayofelony/pwnagotchi#229] [Show and tell]
[Share GPS location from android to pwnagotchi](https://gist.github.com/krishenriksen/87862e2492dcbad4d1a93d869deb1b43)

**Comment by jayofelony** [comment]
Have you tested this?

**Reply by krishenriksen** [reply]
Yes, I made it, proper nmea sentences with checksum calculation.
<img width="1273" alt="Screenshot 2024-11-26 at 11 25 54" src="https://github.com/user-attachments/assets/e9b86c4a-0639-4ffb-bde3-a881712631ec">

**Reply by jayofelony** [reply]
Thats awesome, is there a way of implementing this into the bt-tether plugin as an extra?

**Reply by krishenriksen** [reply]
Very possible, however I cannot make the bluetooth connection stable enough on raspberry pi zero 2 w where as it works just fine on raspberry pi zero w, and I guess most people make pwnagotchi on 2 W now, so it doesn't make sense to allocate more time to this.

On my Raspberry Pi Zero 2 W I have to resort to the following almost every 20 minutes:

sudo lsof | grep bluetooth
sudo systemctl stop NetworkManager
sudo systemctl stop bluetooth
sudo modprobe -r hci_uart
sudo modprobe hci_uart
sudo systemctl start bluetooth
sudo systemctl start NetworkManager
bluetoothctl power on
bluetoothctl connect MAC_ADDRESS
sudo nmcli connection show
sudo nmcli connection modify "A24 Network" ipv4.method manual ipv4.addresses "192.168.44.30/24" ipv4.gateway "192.168.1.1" ipv4.dns "192.168.1.30"
sudo nmcli connection modify "A24 Network" ipv6.method ignore
sudo nmcli connection up "A24 Network"

**Reply by krishenriksen** [reply]
> Thats awesome, is there a way of implementing this into the bt-tether plugin as an extra?

I did it anyway...

https://github.com/jayofelony/pwnagotchi/pull/233

**Comment by krishenriksen** [comment]
Made a small mistake, corrected the invalid coordinates in regards to NMEA 0183 standard.

Also changed the tty device to /dev/ttyUSB0 because bettercap keeps defaulting to this.

**Comment by krishenriksen** [comment]
Updated with VTG 

Bearing and Speed:

    The VTG sentence includes:
        Bearing (true north) in degrees.
        Speed in knots and kilometers per hour.
